torch/nn/modules/transformer.py

```diff
@@ -375,6 +375,100 @@ def forward(self, tgt, memory, tgt_mask=None, memory_mask=None,
         return tgt
 
 
+class MultiheadAttentionInProjection(Module):
+    r"""
+    Args:
+        in_embed_dim (int): Input embedding dimension
+        num_heads (int): Number of parallel attention heads.
+        out_embed_dim (int, optional): Output embedding dimension. If not
+            provided, then set to `in_embed_dim`. Must be divisible by
+            `num_heads`.
+    """
+    def __init__(self, in_embed_dim, num_heads, out_embed_dim=None):
+        # TODO: Bias??
+        super(MultiheadAttentionInProjection, self).__init__()
+        self.in_embed_dim = in_embed_dim
+        if out_embed_dim is None:
+            out_embed_dim = in_embed_dim
+        assert out_embed_dim % num_heads == 0, "out_embed_dim must be divisible by num_heads"
+        self.out_embed_dim = out_embed_dim
+        self.num_heads = num_heads
+        self.weight = Parameter(torch.Tensor(in_embed_dim, out_embed_dim))
+        init.kaiming_uniform_(self.weight, a=math.sqrt(5))
```

Comment:
I basically reused the code from `nn.Linear` for weights and initialization. This offers the flexibility to choose a different initialization. Suggestions?

Reply:
I would not expose this to users for now.

Reply:
I agree. I meant to ask if there is a better initialization to be selecting. It's not a high priority, but it's something I am curious about.